frontend/pages/statistics/index.vue

```diff
@@ -0,0 +1,501 @@
+            }
+          ]
+        },
+        meanAvgTimeText: 0,
+        avgTimeText: {
+          labels: [] as string[],
+          datasets: [
+            {
+              label: 'Chart',
+              backgroundColor: '#f87979',
+              data: [] as number[]
+            }
+          ]
+        }
+      },
+      chartKeys: {
+        weeklyAggregatedStats: {
+          avgTimeAnnotate: 0,
+          avgTimeQuestionnaire: 0,
+          avgTimeText: 0
+        },
+        weeklyIndividualStats: {
+          avgTimeAnnotate: 0,
+          avgTimeQuestionnaire: 0,
+          avgTimeText: 0
+        }
+      }
+    }
+  },
+
+  async fetch() {
+    this.isLoadingUserList = true
+    this.users = await this.$services.user.list(this.username)
+    this.isLoadingUserList = false
+  },
+
+  computed: {
+    weeks() {
+      const startPeriod = moment(this.researchStartDate)
+      const endPeriod = moment()
+      const weeks = []
+      let newStartPeriod = startPeriod
+      let lastDateOfWeek = null
+      let week = ""
+      if (startPeriod.day() !== 1) {
+        const dayPluses = 7 - startPeriod.day()
+        lastDateOfWeek = startPeriod.clone().add(dayPluses, 'days')
+        week = startPeriod.format(this.dateFormatUI) + " - " + lastDateOfWeek.format(this.dateFormatUI)
+        weeks.unshift(week)
+        newStartPeriod = lastDateOfWeek.clone().add(1, 'days')
+      }
+      while (!newStartPeriod.isAfter(endPeriod)) {
+        lastDateOfWeek = newStartPeriod.clone().add(6, 'days')
+        week = newStartPeriod.format(this.dateFormatUI) + " - " + lastDateOfWeek.format(this.dateFormatUI)
+        weeks.unshift(week)
+        newStartPeriod = lastDateOfWeek.clone().add(1, 'days')
+      }
+      return weeks
+    },
+    selectedUsername() {
+      if (this.user !== null) {
+        const user = this.user! as UserDTO
+        return user.username
+      }
+      return ''
+    }
+  },
+
+  watch: {
+    username() {
+      // Items have already been loaded
+      if (this.users.length > 0) return
+      // Items have already been requested
+      if (this.isLoadingUserList) return
+      this.$fetch()
+    },
+    async user(val) {
+      await this.fetchIndividualWeekStatistics(val)
+    }
+  },
+
+  created() {
+    if (!this.isStaff()) {
+      this.$router.push(this.localePath('/projects/'))
+    }
+  },
+
+  async mounted() {
+    this.week = this.weeks[0]
+    this.setWeekStartAndWeekEnd()
+    await this.fetchAggregatedWeekStatistics()
+  },
+
+  methods: {
+    ...mapGetters('auth', ['isStaff']),
+
+    setWeekStartAndWeekEnd() {
+      const weekRange = this.week.split(" - ")
+      this.weekStartDate = weekRange[0]
+      this.weekEndDate = weekRange[1]
+    },
+
+    async weekChangeHandler(week: string) {
+      this.week = week
+      this.setWeekStartAndWeekEnd()
+      await this.fetchAggregatedWeekStatistics()
+      if (this.user !== null){
+        await this.fetchIndividualWeekStatistics(this.user!)
```

Comment:
why is `!` used after `this.user`?

Reply:
I think it is not actually needed because the code `await this.fetchIndividualWeekStatistics(this.user!)` won't be executed if `this.user` is null (i.e. when no individual user is selected). But typescript does not like a variable that can be potentially null, and without the `!` it threw an error that `this.user` is potentially null, even though I have added the check `if (this.user !== null)` above it. Hence, I put the exclamation mark to tell typescript that `this.user` won't be null. 😅

Reply:
I have seen the code and the documentation of `v-autocomplete`, and I can see why TS behaved like that... so I will test it and resolve this later.

Reply:
*if it needs to be resolved, ofc